perception/autoware_scene_to_image_projector/src/autoware_scene_to_image_projector_node.cpp

```diff
@@ -0,0 +1,548 @@
+
+#include "autoware_scene_to_image_projector_node.hpp"
+
+#include <object_recognition_utils/object_recognition_utils.hpp>
+
+#include <boost/optional.hpp>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace
+{
+boost::optional<geometry_msgs::msg::Transform> get_transform_anonymous(
+  const tf2_ros::Buffer & tf_buffer, const std::string & source_frame_id,
+  const std::string & target_frame_id, const rclcpp::Time & time)
+{
+  try {
+    // check if the frames are ready
+    std::string errstr;  // This argument prevents error msg from being displayed in the terminal.
+    if (!tf_buffer.canTransform(
+          target_frame_id, source_frame_id, tf2::TimePointZero, tf2::Duration::zero(), &errstr)) {
+      return boost::none;
+    }
+
+    geometry_msgs::msg::TransformStamped self_transform_stamped;
+    self_transform_stamped = tf_buffer.lookupTransform(
+      /*target*/ target_frame_id, /*src*/ source_frame_id, time,
+      rclcpp::Duration::from_seconds(0.5));
+    return self_transform_stamped.transform;
+  } catch (tf2::TransformException & ex) {
+    RCLCPP_WARN_STREAM(rclcpp::get_logger("multi_object_tracker"), ex.what());
+    return boost::none;
+  }
+}
+
+bool is_polygon_contained(
+  const std::vector<cv::Point2f> & containee,
+  const std::vector<std::vector<cv::Point2f>> & container)
+{
+  for (auto const & poly : container) {
+    size_t counter = 0;
+    std::vector<cv::Point2f> contour;
+    cv::convexHull(poly, contour);
+    for (auto const & point : containee) {
+      if (cv::pointPolygonTest(contour, point, false) >= 0) counter++;
+    }
+    if (containee.size() - counter < 2) return true;
+  }
+  return false;
+}
+
+}  // namespace
+
+namespace autoware::scene_to_image_projector
+{
+SceneToImageProjectorNode::SceneToImageProjectorNode(const rclcpp::NodeOptions & options)
+: Node("scene_to_image_projector", options), tf_buffer_(this->get_clock()), tf_listener_(tf_buffer_)
+{
+  RCLCPP_INFO(this->get_logger(), "SceneToImageProjectorNode::SceneToImageProjectorNode");
+
+  auto objects_type = declare_parameter<std::string>("objects_type", "detected");
+  auto use_trajectory = declare_parameter<bool>("use_trajectory", false);
+  auto use_road_boundaries = declare_parameter<bool>("use_road_boundaries", false);
+
+  show_pedestrian = declare_parameter<bool>("show_pedestrian", true);
+  show_bicycle = declare_parameter<bool>("show_bicycle", true);
+  show_motorcycle = declare_parameter<bool>("show_motorcycle", true);
+  show_trailer = declare_parameter<bool>("show_trailer", true);
+  show_bus = declare_parameter<bool>("show_bus", true);
+  show_truck = declare_parameter<bool>("show_truck", true);
+  show_car = declare_parameter<bool>("show_car", true);
+
+  image_pub_ = this->create_publisher<sensor_msgs::msg::Image>("~/output/image", 10);
+
+  image_sub_ = this->create_subscription<sensor_msgs::msg::Image>(
+    "~/input/image", 10,
```

Comment:
I need to change the QoS to `rclcpp::QoS{10}.best_effort()` here.
Can you check this?